src/main/java/org/truffleruby/core/rope/RopeOperations.java

```diff
@@ -154,7 +154,11 @@ public static String decodeRope(Rope value) {
         value = flatten(value);
 
         Encoding encoding = value.getEncoding();
-        Charset charset = encodingToCharsetMap.computeIfAbsent(encoding, EncodingManager::charsetForEncoding);
+        Charset charset = encodingToCharsetMap.get(encoding);
+        if (charset == null) {
+            charset = EncodingManager.charsetForEncoding(encoding);
+            encodingToCharsetMap.putIfAbsent(encoding, charset);
```

Comment:
Because it's a pure function, I can add that as a comment.

Do you mean just extracting these 5 lines of code in a method?
I cannot easily abstract the call to charsetForEncoding() without overhead.